### PR TITLE
RavenDB-5042

### PR DESCRIPTION
**Pull request by ppekrol**

### CODE REVIEW

File: src/Raven.Server/Config/Categories/MonitoringConfiguration.cs [hunk 1]
```diff
@@ -6,16 +6,17 @@
 
 namespace Raven.Server.Config.Categories
 {
+    [ConfigurationCategory(ConfigurationCategoryType.Monitoring)]
     public class MonitoringConfiguration : ConfigurationCategory
     {
         [Description("A command or executable to run which will provide machine cpu usage and total machine cores to standard output. If specified, RavenDB will use this information for monitoring cpu usage.")]
         [DefaultValue(null)]
-        [ConfigurationEntry("Monitoring.Cpu.Exec", ConfigurationEntryScope.ServerWideOnly)]
+        [ConfigurationEntry("Monitoring.Cpu.Exec", ConfigurationEntryScope.ServerWideOnly, isSecured: true)]
```
Comment (ayende) [inline src/Raven.Server/Config/Categories/MonitoringConfiguration.cs]: Why is this secured?

File: src/Raven.Server/Config/ConfigurationEntryValue.cs [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+            {
+                var value = configuration.GetServerWideSetting(key);
+                if (value == null)
+                    continue;
+
+                var hasAccess = authenticationStatus == RavenServer.AuthenticationStatus.ClusterAdmin;
+                ServerValues[key] = new ConfigurationEntrySingleValue
+                {
+                    Value = hasAccess ? value : null,
+                    HasAccess = hasAccess
```
Comment (ayende) [inline src/Raven.Server/Config/ConfigurationEntryValue.cs:42]: Where is this being used?

File: src/Raven.Server/Config/Attributes/ConfigurationEntryAttribute.cs [hunk 3]
```diff
@@ -12,11 +12,12 @@ namespace Raven.Server.Config.Attributes
     [AttributeUsage(AttributeTargets.Property, AllowMultiple = true)]
     public class ConfigurationEntryAttribute : Attribute
     {
-        public ConfigurationEntryAttribute(string key, ConfigurationEntryScope scope, [CallerLineNumber]int order = 0, bool setDefaultValueIfNeeded = true) // the default order is the order of declaration in a configuration class
+        public ConfigurationEntryAttribute(string key, ConfigurationEntryScope scope, [CallerLineNumber]int order = 0, bool setDefaultValueIfNeeded = true, bool isSecured = false) // the default order is the order of declaration in a configuration class
         {
             Key = key;
             Order = order;
             SetDefaultValueIfNeeded = setDefaultValueIfNeeded;
+            IsSecured = isSecured;
```
Comment (ayende) [inline src/Raven.Server/Config/Attributes/ConfigurationEntryAttribute.cs:20]: I don't understand the purpose of this? Are we talking about private information? Such as passwords?

File: src/Raven.Server/Config/ConfigurationEntryValue.cs (hunk 2 above)
Comment (ppekrol) [reply src/Raven.Server/Config/ConfigurationEntryValue.cs:42]: It is for the UI, so we will be able to write there that the value is set, but user do not have access to it (e.g. DatabaseAdmin should not see ServerWideOnly values)

File: src/Raven.Server/Config/Attributes/ConfigurationEntryAttribute.cs (hunk 3 above)
Comment (ppekrol) [reply src/Raven.Server/Config/Attributes/ConfigurationEntryAttribute.cs:20]: It is for the UI, so we can substitute the value with `***********` with 'button' to see the real value

File: src/Raven.Server/Config/Categories/MonitoringConfiguration.cs (hunk 1 above)
Comment (ppekrol) [reply src/Raven.Server/Config/Categories/MonitoringConfiguration.cs]: I've secured all 'exec' values

File: src/Raven.Server/Config/ConfigurationEntryValue.cs (hunk 2 above)
Comment (ayende) [reply src/Raven.Server/Config/ConfigurationEntryValue.cs:42]: Where are you doing the filtering?

File: src/Raven.Server/Config/Attributes/ConfigurationEntryAttribute.cs (hunk 3 above)
Comment (ayende) [reply src/Raven.Server/Config/Attributes/ConfigurationEntryAttribute.cs:20]: Why? The arguments, I can understand, but the exec doesn't seems like it should be.

File: src/Raven.Server/Config/Categories/SecurityConfiguration.cs [hunk 4]
```diff
@@ -10,160 +10,161 @@
 
 namespace Raven.Server.Config.Categories
 {
+    [ConfigurationCategory(ConfigurationCategoryType.Security)]
     public class SecurityConfiguration : ConfigurationCategory
     {
         [Description("Disable automatic redirection when listening to HTTPS. By default, when using port 443, RavenDB redirects all incoming HTTP traffic on port 80 to HTTPS on port 443.")]
         [DefaultValue(false)]
-        [ConfigurationEntry("Security.DisableHttpsRedirection", ConfigurationEntryScope.ServerWideOnly)]
+        [ConfigurationEntry("Security.DisableHttpsRedirection", ConfigurationEntryScope.ServerWideOnly, isSecured: true)]
```
Comment (ayende) [inline src/Raven.Server/Config/Categories/SecurityConfiguration.cs]:
```suggestion
        [ConfigurationEntry("Security.DisableHttpsRedirection", ConfigurationEntryScope.ServerWideOnly, isSecured: false)]
```

File: src/Raven.Server/Config/Categories/SecurityConfiguration.cs [hunk 5]
```diff
@@ -10,160 +10,161 @@
...
         public bool DisableHttpsRedirection { get; set; }
 
         [Description("The path to a folder where RavenDB will store the access audit logs")]
         [DefaultValue(null)]
-        [ConfigurationEntry("Security.AuditLog.FolderPath", ConfigurationEntryScope.ServerWideOnly)]
+        [ConfigurationEntry("Security.AuditLog.FolderPath", ConfigurationEntryScope.ServerWideOnly, isSecured: true)]
```
Comment (ayende) [inline src/Raven.Server/Config/Categories/SecurityConfiguration.cs]: Not secret

File: src/Raven.Server/Config/Categories/SecurityConfiguration.cs [hunk 6]
```diff
@@ -10,160 +10,161 @@
...
         public PathSetting AuditLogPath { get; set; }
 
         [Description("How far back we should retain audit log entries")]
         [DefaultValue(365 * 24)]
         [TimeUnit(TimeUnit.Hours)]
-        [ConfigurationEntry("Security.AuditLog.RetentionTimeInHrs", ConfigurationEntryScope.ServerWideOnly)]
-        [ConfigurationEntry("Security.AuditLog.RetentionTimeInHours", ConfigurationEntryScope.ServerWideOnly)]
+        [ConfigurationEntry("Security.AuditLog.RetentionTimeInHrs", ConfigurationEntryScope.ServerWideOnly, isSecured: true)]
+        [ConfigurationEntry("Security.AuditLog.RetentionTimeInHours", ConfigurationEntryScope.ServerWideOnly, isSecured: true)]
         public TimeSetting AuditLogRetentionTime { get; set; }
```
Comment (ayende) [inline src/Raven.Server/Config/Categories/SecurityConfiguration.cs]: Not secret

File: src/Raven.Server/Config/Categories/SecurityConfiguration.cs [hunk 7]
```diff
@@ -10,160 +10,161 @@
...
 
         [Description("The maximum size of the audit log after which the old files will be deleted")]
         [DefaultValue(null)]
         [MinValue(256)]
         [SizeUnit(SizeUnit.Megabytes)]
-        [ConfigurationEntry("Security.AuditLog.RetentionSizeInMb", ConfigurationEntryScope.ServerWideOnly)]
+        [ConfigurationEntry("Security.AuditLog.RetentionSizeInMb", ConfigurationEntryScope.ServerWideOnly, isSecured: true)]
```
Comment (ayende) [inline src/Raven.Server/Config/Categories/SecurityConfiguration.cs]: A lot of those values are not secrets.
Basically, just passwords should be hidden

File: src/Raven.Server/Config/ConfigurationEntryValue.cs (hunk 2 above)
Comment (ppekrol) [reply src/Raven.Server/Config/ConfigurationEntryValue.cs:42]: There was no intention to do any filtering. It depends how we will design the UI.

File: src/Raven.Server/Config/Attributes/ConfigurationEntryAttribute.cs (hunk 3 above)
Comment (ppekrol) [reply src/Raven.Server/Config/Attributes/ConfigurationEntryAttribute.cs:20]: Depends on the angle. I prefer to hide it, because it reveals a path.

File: src/Raven.Server/Config/ConfigurationEntryValue.cs (hunk 2 above)
Comment (ayende) [reply src/Raven.Server/Config/ConfigurationEntryValue.cs:42]: Let me rephrase, where are you replacing the value with `****`

Comment (ppekrol) [reply src/Raven.Server/Config/ConfigurationEntryValue.cs:42]: Done

File: src/Raven.Server/Config/Attributes/ConfigurationEntryAttribute.cs (hunk 3 above)
Comment (ppekrol) [reply src/Raven.Server/Config/Attributes/ConfigurationEntryAttribute.cs:20]: Done

**Comment by ppekrol** [comment]
@ayende I've applied your comments. Removed isSecured from most of the configuration entries. Left few. Could you look at them?

File: src/Raven.Server/Config/Categories/SecurityConfiguration.cs [hunk 8]
```diff
@@ -104,12 +105,12 @@ public class SecurityConfiguration : ConfigurationCategory
 
         [Description("The E-mail address associated with the Let's Encrypt certificate. Used for renewal requests.")]
         [DefaultValue(null)]
-        [ConfigurationEntry("Security.Certificate.LetsEncrypt.Email", ConfigurationEntryScope.ServerWideOnly)]
+        [ConfigurationEntry("Security.Certificate.LetsEncrypt.Email", ConfigurationEntryScope.ServerWideOnly, isSecured: true)]
```
Comment (ayende) [inline src/Raven.Server/Config/Categories/SecurityConfiguration.cs]: Not a secret, probably

File: src/Raven.Server/Config/Categories/SecurityConfiguration.cs [hunk 9]
```diff
@@ -104,12 +105,12 @@ public class SecurityConfiguration : ConfigurationCategory
...
         public string CertificateLetsEncryptEmail { get; set; }
 
         [Description("The path of the (256-bit) Master Key. If specified, RavenDB will use this key to protect secrets.")]
         [DefaultValue(null)]
-        [ConfigurationEntry("Security.MasterKey.Path", ConfigurationEntryScope.ServerWideOnly)]
+        [ConfigurationEntry("Security.MasterKey.Path", ConfigurationEntryScope.ServerWideOnly, isSecured: true)]
```
Comment (ayende) [inline src/Raven.Server/Config/Categories/SecurityConfiguration.cs]: Why is that a secret? It convey no additional information

File: src/Raven.Server/Config/Categories/SecurityConfiguration.cs [hunk 10]
```diff
@@ -135,12 +136,12 @@ public class SecurityConfiguration : ConfigurationCategory
...
         [DefaultValue(null)]
-        [ConfigurationEntry("Security.WellKnownCertificates.Admin", ConfigurationEntryScope.ServerWideOnly)]
+        [ConfigurationEntry("Security.WellKnownCertificates.Admin", ConfigurationEntryScope.ServerWideOnly, isSecured: true)]
         public string[] WellKnownAdminCertificates { get; set; }
 
         [Description("Well known issuer 'Public Key Pinning Hashes' that will be used to validate a new client certificate when the issuer's certificate has changed.")]
         [DefaultValue(null)]
-        [ConfigurationEntry("Security.WellKnownIssuerHashes.Admin", ConfigurationEntryScope.ServerWideOnly)]
+        [ConfigurationEntry("Security.WellKnownIssuerHashes.Admin", ConfigurationEntryScope.ServerWideOnly, isSecured: true)]
         public string[] WellKnownIssuerHashes { get; set; }
```
Comment (ayende) [inline src/Raven.Server/Config/Categories/SecurityConfiguration.cs]: Not a secret (and we report that to user anyway)

File: src/Raven.Server/Config/Categories/SecurityConfiguration.cs [hunk 11]
```diff
@@ -135,12 +136,12 @@ public class SecurityConfiguration : ConfigurationCategory
 
         [Description("Well known certificate thumbprints that will be trusted by the server as cluster admins.")]
         [DefaultValue(null)]
-        [ConfigurationEntry("Security.WellKnownCertificates.Admin", ConfigurationEntryScope.ServerWideOnly)]
+        [ConfigurationEntry("Security.WellKnownCertificates.Admin", ConfigurationEntryScope.ServerWideOnly, isSecured: true)]
         public string[] WellKnownAdminCertificates { get; set; }
```
Comment (ayende) [inline src/Raven.Server/Config/Categories/SecurityConfiguration.cs]: Not a secret (and we report that to user anyway)

**Review by ayende — APPROVED** [review]
